### PR TITLE
docs: TI review of wiki pages and updates to readme files

### PR DESCRIPTION
**Pull request by zeibura**
Not finished yet, creating this so that I can keep track of work done so far.
- Docs files up to "home.md" are finished. I'm not sure if "home.MD" needs to be proofread as it is still formatted in MediaWiki rather than in Markdown. To resume tomorrow from "libraries configuration"
- Some changes have been made to the new readme content and the readme structure (the bottom section split into two subsections and renamed configuring, as hinted by the link in a section above)
- A few minor changes to Changelog.MD

### CODE REVIEW

**Comment by michalbali256** [comment]
> I'm not sure if "home.MD" needs to be proofread as it is still formatted in MediaWiki rather than in Markdown.

The whole documentation is published on [GitHub wiki](https://github.com/eclipse/che-che4z-lsp-for-hlasm/wiki), for which are the MediaWiki references preferable over markdown references (I am not 100% sure, but I think markdown references required full url to target).

**Comment by zeibura** [comment]
> The whole documentation is published on [GitHub wiki](https://github.com/eclipse/che-che4z-lsp-for-hlasm/wiki), for which are the MediaWiki references preferable over markdown references (I am not 100% sure, but I think markdown references required full url to target).

Correct about links. However the rest of the pages are formatted in markdown, only home.MD is in mediawiki (and the mediawiki links don't show up properly)

**Comment by michalbali256** [comment]
All pages are formatted in markdown, only links between the pages are written using the MediaWiki syntax. I agree it may be a bit messy but at least for me it easy to use (mainly because I do not know MW syntax).

> and the mediawiki links don't show up properly

They don't show up properly when viewing the sources directly, but they show up properly in GitHub wiki - that is the target platform. The `docs/` folder is automatically published to wiki after each PR merge.

**Comment by zeibura** [comment]
OK, thanks for the explanation, I wasn't aware the GitHub wiki could read both formats at once

**Comment by zeibura** [comment]
All files in the root folder are now done
One rather high level point is that the page title "libraries configuration" should be "configuration of libraries". I will move the page as it would be done in MediaWiki, leaving behind a redirect, just in case any links to the old page remain anywhere.

**Comment by michalbali256** [comment]
Sure, go ahead with the renaming. I don't see any problems it will cause.

**Comment by zeibura** [comment]
All pages are done

File: docs/Language-server-pages/LSP-and-DAP-server.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
 LSP and DAP Server
 ------------------
```
Comment (michalbali256) [inline docs/Language-server-pages/LSP-and-DAP-server.md:2]:
```suggestion
```

File: docs/Language-server-pages/Request-manager.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Request Manager
 ---------------
```
Comment (michalbali256) [inline docs/Language-server-pages/Request-manager.md:2]:
```suggestion
```

**Comment by zeibura** [comment]
Thansk! Just FYI, MediaWiki pipe links are [[target|link text]], not [[link text|target]]. I fixed one more.

**Pull request merged**